src/main/java/jsr223/nativeshell/NativeShellRunner.java

```diff
@@ -81,19 +112,35 @@ public int run(String command, ScriptContext scriptContext) throws ScriptExcepti
         }
     }
 
-    private int run(File command, ScriptContext scriptContext) throws ScriptException{
+    private int run(File command, ScriptContext scriptContext) throws ScriptException {
         ProcessBuilder processBuilder = nativeShell.createProcess(command);
 
         processBuilder.environment();
 
         Map<String, String> environment = processBuilder.environment();
+        List<String> arguments = getArguments(scriptContext);
         addBindingsAsEnvironmentVariables(scriptContext, environment);
+        processBuilder.command().addAll(arguments);
         CookieBasedProcessTreeKiller processTreeKiller = createProcessTreeKiller(scriptContext, environment);
 
         return run(processBuilder,
                    scriptContext.getReader(),
                    scriptContext.getWriter(),
-                   scriptContext.getErrorWriter(), processTreeKiller);
+                   scriptContext.getErrorWriter(),
+                   processTreeKiller);
+    }
+
+    private List<String> getArguments(ScriptContext scriptContext) {
+        Bindings bindings = scriptContext.getBindings(ScriptContext.ENGINE_SCOPE);
+        if (bindings != null && bindings.containsKey(ARGS)) {
+            if (bindings.get(ARGS) instanceof String[]) {
+                String[] arguments = (String[]) bindings.get(ARGS);
+                return Arrays.asList(arguments);
```

Comment:
just to confirm, `Arrays.asList` returns array which pretends to be a List

Reply:
It does not matter, as long as it is a collection which can be added to the command